src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -972,12 +969,17 @@ private Task WriteSuffix()
                 responseHeaders.SetRawDate(dateHeaderValues.String, dateHeaderValues.Bytes);
             }
 
-            var writableBuffer = Output.Alloc();
+            Output.Write(_writeHeaders, this);
+        }
+
+        private static void WriteResponseHeaders(WritableBuffer writableBuffer, Frame frame)
+        {
+            var responseHeaders = frame.FrameResponseHeaders;
             writableBuffer.WriteFast(_bytesHttpVersion11);
+            var statusBytes = ReasonPhrases.ToStatusBytes(frame.StatusCode, frame.ReasonPhrase);
             writableBuffer.WriteFast(statusBytes);
             responseHeaders.CopyTo(ref writableBuffer);
             writableBuffer.WriteFast(_bytesEndHeaders);
```

Comment:
`frame._bytesEndHeaders`

Doesn't matter much though (except for consistency) since you're not really creating a closure.